test/test_composability.py

```diff
@@ -229,25 +232,32 @@ def build_stage(stage_idx, num_stages):
             )
             partial_model.to(device)
 
-            # apply FSDP
-            mp_policy = MixedPrecisionPolicy(
-                # TODO(whc) need to fix PP + FSDP-mixed-precision
-                # tracer for PP assumes f32 and is caught off guard when runtime FSDP interacts using bf16 inputs
-                # param_dtype=torch.bfloat16, reduce_dtype=torch.float32
-                param_dtype=torch.float32,
-                reduce_dtype=torch.float32,
-            )
-            fsdp_config = {"mesh": dp_mesh, "mp_policy": mp_policy}
-            for layer in partial_model.children():
-                fully_shard(
-                    layer,
-                    **fsdp_config,
-                    reshard_after_forward=False,
+            if dp_type == "FSDP":
```

Comment:
nit: maybe separate out the two cases into two tests? One for PP + FSDP and one for PP + DDP.

nit: maybe move the appliance of FSDP/DP out of `build_stage`? Closer to titan style.